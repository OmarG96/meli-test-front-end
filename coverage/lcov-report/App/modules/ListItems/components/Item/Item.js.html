<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for App/modules/ListItems/components/Item/Item.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\..\..\index.html">All files</a> / <a href="index.html">App/modules/ListItems/components/Item</a> Item.js
    </h1>
    <div class='clearfix'>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { useCallback } from "react";
import { bool, number, oneOfType, shape, string } from "prop-types";
&nbsp;
import { formatMoney } from "../../../../../utils/methods/formatMoney";
&nbsp;
import iconShipping from "../../../../../assets/images/ic_shipping.png";
import "./style.scss";
import { useHistory } from "react-router-dom";
&nbsp;
const Item = ({ data }) =&gt; {
  const history = useHistory();
  const handleClick = useCallback(() =&gt; {
    history.push(`/items/${data.id}`);
  }, [history]);
&nbsp;
  return (
    &lt;div role="item" className="item-root" onClick={handleClick}&gt;
      &lt;div className="item"&gt;
        &lt;div className="picture"&gt;
          &lt;img src={data.picture} /&gt;
        &lt;/div&gt;
        &lt;div className="detail"&gt;
          &lt;div className="header"&gt;
            &lt;div className="price"&gt;
              {formatMoney(data.price.amount)}
              {data.free_shipping ? (
                &lt;img className="icon-shipping" src={iconShipping} /&gt;
              ) : null}
            &lt;/div&gt;
            &lt;div className="state-name"&gt;{data.state_name}&lt;/div&gt;
          &lt;/div&gt;
          &lt;div className="title"&gt;{data.title}&lt;/div&gt;
        &lt;/div&gt;
      &lt;/div&gt;
    &lt;/div&gt;
  );
};
&nbsp;
Item.propTypes = {
  data: shape({
    id: string,
    title: string,
    price: shape({
      currency: string,
      amount: oneOfType([string, number]),
    }),
    picture: string,
    condition: string,
    free_shipping: bool,
    sold_quantity: oneOfType([string, number]),
    description: string,
  }),
};
&nbsp;
export default Item;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Mon Jun 21 2021 18:32:13 GMT-0300 (GMT-03:00)
</div>
</div>
<script src="..\..\..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\..\..\sorter.js"></script>
</body>
</html>
