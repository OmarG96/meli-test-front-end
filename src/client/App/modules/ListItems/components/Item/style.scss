@import "../../../../../utils/scss/variables.scss";

.item-root {
  padding: 0 16px;
  cursor: pointer;

  &:not(:last-child) {
    .item {
      border-bottom: 1px solid $gray-1;
    }
  }
}

.item {
  display: flex;
  padding: 16px 0;
}

.picture {
  margin-right: 16px;
  position: relative;
  display: block;
  width: 180px;
  height: 180px;
  border-radius: 4px;

  img {
    display: block;
    object-fit: contain;
    width: 100%;
    height: 100%;
  }
}

.detail {
  flex: 5;

  display: flex;
  flex-direction: column;
}

.header {
  height: 48px;
  display: flex;
  align-items: flex-end;
  margin-bottom: 32px;
}

.price {
  flex: 5;
  font-size: 24px;
  color: $gray-3;
}

.icon-shipping {
  margin-left: 12px;
}

.state-name {
  flex: 2;
  width: 100%;
  color: $gray-6;
  font-size: 12px;
}

.title {
  font-size: 18px;
  color: $gray-3;
  max-width: 400px;
}
