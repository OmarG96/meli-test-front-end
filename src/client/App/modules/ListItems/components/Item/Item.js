import React, { useCallback } from "react";
import { bool, number, oneOfType, shape, string } from "prop-types";

import { formatMoney } from "../../../../../utils/methods/formatMoney";

import iconShipping from "../../../../../assets/images/ic_shipping.png";
import "./style.scss";
import { useHistory } from "react-router-dom";

const Item = ({ data }) => {
  const history = useHistory();
  const handleClick = useCallback(() => {
    history.push(`/items/${data.id}`);
  }, [history]);

  return (
    <div role="item" className="item-root" onClick={handleClick}>
      <div className="item">
        <div className="picture">
          <img src={data.picture} />
        </div>
        <div className="detail">
          <div className="header">
            <div className="price">
              {formatMoney(data.price.amount)}
              {data.free_shipping ? (
                <img className="icon-shipping" src={iconShipping} />
              ) : null}
            </div>
            <div className="state-name">{data.state_name}</div>
          </div>
          <div className="title">{data.title}</div>
        </div>
      </div>
    </div>
  );
};

Item.propTypes = {
  data: shape({
    id: string,
    title: string,
    price: shape({
      currency: string,
      amount: oneOfType([string, number]),
    }),
    picture: string,
    condition: string,
    free_shipping: bool,
    sold_quantity: oneOfType([string, number]),
    description: string,
  }),
};

export default Item;
