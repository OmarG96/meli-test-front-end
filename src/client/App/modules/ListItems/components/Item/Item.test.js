import React from "react";
import Item from ".";
import { fireEvent, render, screen, waitFor } from "@testing-library/react";

const mockPush = jest.fn();

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useHistory: () => ({
    push: mockPush,
  }),
}));

const dummyData = {
  id: "EXAMPLE123",
  title: "Title title title",
  price: {
    currency: "ARS",
    amount: "123456",
  },
  picture: "url.example.com",
  condition: "new",
  free_shipping: true,
  sold_quantity: 123,
  description: "description description description. Description.",
};

describe("GIVEN Item", () => {
  describe("WHEN is rendered", () => {
    it("THEN match to snapshot", async () => {
      const { asFragment } = render(<Item data={dummyData} />);
      await screen.findByText("Title title title");
      expect(asFragment()).toMatchSnapshot();
    });

    it("THEN match to snapshot when free shipping is false", async () => {
      const { asFragment } = render(
        <Item data={{ ...dummyData, free_shipping: false }} />
      );
      await screen.findByText("Title title title");
      expect(asFragment()).toMatchSnapshot();
    });

    it("THEN redirect to /items/EXAMPLE123 when is clicked", async () => {
      render(<Item data={dummyData} />);
      await waitFor(() => fireEvent.click(screen.getByRole("item")));

      expect(mockPush).toBeCalledWith("/items/EXAMPLE123");
    });
  });
});
