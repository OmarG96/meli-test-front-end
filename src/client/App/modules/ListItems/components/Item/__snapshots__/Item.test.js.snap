// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GIVEN Item WHEN is rendered THEN match to snapshot 1`] = `
<DocumentFragment>
  <div
    class="item-root"
    role="item"
  >
    <div
      class="item"
    >
      <div
        class="picture"
      >
        <img
          src="url.example.com"
        />
      </div>
      <div
        class="detail"
      >
        <div
          class="header"
        >
          <div
            class="price"
          >
            $ 123,456
            <img
              class="icon-shipping"
              src="test-file-stub"
            />
          </div>
          <div
            class="state-name"
          />
        </div>
        <div
          class="title"
        >
          Title title title
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`GIVEN Item WHEN is rendered THEN match to snapshot when free shipping is false 1`] = `
<DocumentFragment>
  <div
    class="item-root"
    role="item"
  >
    <div
      class="item"
    >
      <div
        class="picture"
      >
        <img
          src="url.example.com"
        />
      </div>
      <div
        class="detail"
      >
        <div
          class="header"
        >
          <div
            class="price"
          >
            $ 123,456
          </div>
          <div
            class="state-name"
          />
        </div>
        <div
          class="title"
        >
          Title title title
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
